argilla/docs/reference/argilla-server/oauth2_configuration.md

```diff
@@ -0,0 +1,134 @@
+# OAuth2 configuration
+
+Argilla supports OAuth2 authentication for users. This allows users to authenticate using other services like Google, GitHub, or Hugging Face.
+Next sections will guide you through the configuration of the OAuth2 authentication.
+
+## The OAuth2 configuration file
+
+The OAuth2 configuration file is a YAML file that contains the configuration for the OAuth2 providers that you want to enable.
+The default file name is `.oauth.yml` and it should be placed in the root directory of the Argilla server. You can also specify a different file name using the `ARGILLA_OAUTH_CONFIG_FILE` environment variable.
+
+The file should have the following structure:
+
+```yaml
+providers:
+  - name: huggingface
+    client_id: "<client_id>"
+    client_secret: "<client_secret>"
+    scope: "openid profile"
+
+  - name: google-oauth2
+    client_id: "<client_id>"
+    client_secret: "<client_secret>"
+    scope: "openid email profile"
+
+  - name: github
+    client_id: "<client_id>"
+    client_secret: "<client_secret>"
+
+allowed_workspaces:
+  - name: argilla
+
+allow_http_redirect: false
+```
+
+### Providers
+
+The `providers` key is a list of dictionaries, each dictionary represents a provider configuration, including the following fields:
+
+- `name`: The name of the provider. The available options by default are `huggingface`, `github` and `google-oauth2`. We will see later how to add more providers not supported by default.
+- `client_id`: The client ID provided by the OAuth2 provider. You can get this value by creating an application in the provider's developer console. This is a required field, but you can also use the `ARGILLA_OAUTH2_<PROVIDER_NAME>_CLIENT_ID` environment variable to set the value.
+- `client_secret`: The client secret provided by the OAuth2 provider. You can get this value by creating an application in the provider's developer console. This is a required field, but you can also use the `ARGILLA_OAUTH2_<PROVIDER_NAME>_CLIENT_SECRET` environment variable to set the value.
+- `scope`: The scope of the OAuth2 provider. This is an optional field, and normally you don't need to set it, but you can use it to request specific permissions from the user access.
+
+### Allowed Workspaces
+
+The `allowed_workspaces` key is a list of dictionaries, each dictionary represents a workspace configuration. For now, only the `name` field is defined, and it should be the name of the workspace that the user will be assigned to when logging in.
```

Comment:
what do you mean by "a list of dictionaries"? It seems the we just provide a list of `-name` right? or are we missing some info her"

Reply:
what happens if we don't use allowed workspaces? will it fail or will people just get access to their own workspaces?

Reply:
So, people with`write` access to an organisation to get the `owner` role and `admin` access the `admin` role? 

How do we deal with access from people that are not part of the organisation?

I think not providing a workspace should still allow people to get access to their own workspace?